### PR TITLE
Make travis run clang tests in parallel

### LINKED ISSUE #345: Openmp on linux with clang generates incorrect results
After installing omp for clang, when I use compiled executables I'm getting results that differ depending on the order of declarations, or the use of unrelated declarations.

With this dl file, changing the order of B and C, or removing A, changes the result
```
.decl A()

.decl B()
.output B
B().
B() :- C().

.decl C()
.output C()
C() :- B().
```
I found the executable was getting lost in this block of code from the generated cpp file
```
  SECTIONS_START;
  SECTION_START;
  if (!rel_5_delta_C->empty()) {
** Not reached
    [&]() {
** 2
      CREATE_OP_CONTEXT(rel_5_delta_C_op_ctxt, rel_5_delta_C->createContext());
      CREATE_OP_CONTEXT(rel_3_new_B_op_ctxt, rel_3_new_B->createContext());
      CREATE_OP_CONTEXT(rel_1_B_op_ctxt, rel_1_B->createContext());
      if (!rel_5_delta_C->empty()) {
        if (!rel_1_B->contains(Tuple<RamDomain, 0>({}), READ_OP_CONTEXT(rel_1_B_op_ctxt))) {
          Tuple<RamDomain, 0> tuple({});
          rel_3_new_B->insert(tuple, READ_OP_CONTEXT(rel_3_new_B_op_ctxt));
        }
      }
    }();
** Not reached
  }
  SECTION_END
  SECTION_START;
  if (!rel_2_delta_B->empty()) {
** 1
    [&]() {
      CREATE_OP_CONTEXT(rel_2_delta_B_op_ctxt, rel_2_delta_B->createContext());
      CREATE_OP_CONTEXT(rel_6_new_C_op_ctxt, rel_6_new_C->createContext());
      CREATE_OP_CONTEXT(rel_4_C_op_ctxt, rel_4_C->createContext());
      if (!rel_2_delta_B->empty()) {
        if (!rel_4_C->contains(Tuple<RamDomain, 0>({}), READ_OP_CONTEXT(rel_4_C_op_ctxt))) {
          Tuple<RamDomain, 0> tuple({});
          rel_6_new_C->insert(tuple, READ_OP_CONTEXT(rel_6_new_C_op_ctxt));
        }
      }
    }();
** 3
  }
  SECTION_END
  SECTIONS_END;

```
Execution skipped the first block, since C is empty to begin with, reaches (1), jumped to the lambda in the first block (2), then once that was done jumped back to the second block after its lambda (3). g++ worked as I expected, as did clang++ without omp.

It would be nice if correct results were returned with clang/omp enabled.

### PR DESCRIPTION
**Pull request by mmcgr**
Make travis run clang tests in parallel
Refactor setup scripts - they were getting hard to follow with the new changes
Fixes #345, or at least works around the problem.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -14,30 +14,37 @@ compiler:
 env:
   - TEST_FORMAT=1
   - MAKEPACKAGE=1
-  - SOUFFLE_CATEGORY=Unit,Syntactic,Semantic,Interface,Profile
-  - SOUFFLE_CATEGORY=Evaluation SOUFFLE_CONFS=",-j8,-p profile.log"
```
Comment (scyptnex) [inline .travis.yml]: you missed the empty flags test

File: .travis.yml [hunk 2]
```diff
@@ -14,30 +14,37 @@ compiler:
...
+    env: SOUFFLE_CATEGORY=Evaluation SOUFFLE_CONFS="-j8,-c -j8"
+  - os: linux
     compiler: clang
-    env: MAKEPACKAGE=1 
+    env: MAKEPACKAGE=1
   - os: osx
     compiler: gcc
 
-script: 
+script:
```
Comment (scyptnex) [inline .travis.yml]: I dont really mind, but potentially you want the travis "before_script" for this

File: .travis.yml (hunk 1 above)
Comment (mmcgr) [inline .travis.yml]: Done

File: .travis.yml (hunk 2 above)
Comment (mmcgr) [inline .travis.yml]: Done. Also, much neater now.
